Short-Answer/Algorithms_Answers.md

```diff
@@ -1 +1,58 @@
-Add your answers to the Algorithms exercises here.
\ No newline at end of file
+# Analysis of Algorithms
+
+## Exercise I
+
+```
+a)  a = 0                                           O(1)
+    while (a < n * n * n):                          O(n^3)
+      a = a + n * n                                 O(1)
+```
+
+Calcs:
+O(1) + O(n^3 * 1)
+### O(n^3)
```

Comment:
```
a)  a = 0                      O(1)
    while (a < n * n * n):     normally O(n^3), but...
      a = a + n * n            here we add n^2 to `a` each time
```

The whole thing is similar to for loop:

```
for (a = 0; a < n * n * n; a += n * n)
```

Those `n`s effectively cancel out to:

```
for (a = 0; a < n; a += 1)     Clearly O(n)
```

Answer: `O(n)`